### PR TITLE
[WFLY-19049] Use testcontainers to start Mail server for running mail…

### PR DESCRIPTION
**Pull request by yersan**
… tests

Jira issue: https://issues.redhat.com/browse/WFLY-19049

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/mail/basic/MailTestCase.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+            // POP3 Mail server
+            op = Util.createAddOperation(PathAddress.parseCLIStyleAddress("/subsystem=mail/mail-session=mail-test-basic/server=pop3"));
+            op.get("outbound-socket-binding-ref").set("mail-test-basic-pop3-binding");
+
+            ManagementOperations.executeOperation(managementClient.getControllerClient(), op);
+
+            // Reload to apply the changes
+            ServerReload.reloadIfRequired(managementClient);
+        }
```
Comment (rhusar) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/mail/basic/MailTestCase.java:115]: Just use `org.jboss.as.test.shared.ManagementServerSetupTask` instead.

**Review by rhusar — CHANGES_REQUESTED** [review]

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/mail/basic/MailTestCase.java (hunk 1 above)
Comment (yersan) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/mail/basic/MailTestCase.java:115]: Ok, got it, thanks @rhusar . The operations don't need to go in a batch. Out of curiosity, what would be the advantage vs the current one?

Comment (yersan) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/mail/basic/MailTestCase.java:115]: Well, I guess, they are applied faster, any other?

Comment (yersan) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/mail/basic/MailTestCase.java:115]: ahh, ok, I see it, if one of them fails, there won't be inconsistencies with partially applied operations, that's good

**Comment by bstansberry** [comment]
@rhusar is this ok now?

**Review by rhusar — APPROVED** [review]
LGTM now, cool stuff!

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/mail/basic/MailTestCase.java (hunk 1 above)
Comment (rhusar) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/mail/basic/MailTestCase.java]: Sorry forgot to reply, but yes apart from getting a consistent server config, it's also simple to use and less verbose = less LOC to manage :)

**Comment by bstansberry** [comment]
Thanks @yersan and @rhusar

**Pull request merged**